src/components/FullArticleCard.jsx

```diff
@@ -1,19 +1,66 @@
 import "bootstrap/dist/css/bootstrap.min.css";
 import Button from "react-bootstrap/Button";
 import Card from "react-bootstrap/Card";
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { patchArticle } from "../api";
 
 function FullArticleCard({ article }) {
-  // console.log(article, "<--article inside fullArticleCard");
+  const actualVoteCount = article.votes;
+
+  const [displayedVoteCount, setDisplayedVoteCount] = useState(actualVoteCount);
+  const [disabledButtonsLogic, setdisabledButtonsLogic] = useState([
+    false,
+    false,
+  ]); //i.e. both buttons are initially not disabled
+
+  useEffect(() => {
+    setDisplayedVoteCount(actualVoteCount);
+  }, [actualVoteCount]);
+
+  function handleClick(article_id, increment, button_index) {
+    patchArticle(article_id, increment);
+    const incrementedVoteCount = displayedVoteCount + increment;
+
+    setDisplayedVoteCount(incrementedVoteCount);
+
+    //set button states - user should not be able to incrmenet the vote count by more than one in either direction
+    let disabledButtonsLogicCopy = [...disabledButtonsLogic];
+    if (
+      (incrementedVoteCount - actualVoteCount < 0 && button_index === 1) ||
+      (incrementedVoteCount - actualVoteCount > 0 && button_index === 0)
+    ) {
+      disabledButtonsLogicCopy[button_index] = true;
+    } else {
+      disabledButtonsLogicCopy = [false, false];
+    }
+
+    setdisabledButtonsLogic(disabledButtonsLogicCopy);
+  }
   return (
     <div className="fullArticleCard">
       <Card>
-        <Card.Title><h1>{article.title}</h1></Card.Title>
+        <Card.Title>
+          <h1>{article.title}</h1>
+        </Card.Title>
         <Card.Subtitle>by {article.author}</Card.Subtitle>
         <Card.Body>
           <Card.Img src={article.article_img_url} variant="top" />
           <Card.Text className="articleBody">{article.body}</Card.Text>
-          <Card.Text>{article.votes} votes</Card.Text>
+          <Button
+            onClick={() => handleClick(article.article_id, 1, 0)}
```

Comment:
This is nice - reusing the handle click with different values instead of 2 separate function 👍 Also disabling the buttons works well.